src/Development/IDE/Core/Rules.hs

```diff
@@ -500,6 +502,25 @@ typeCheckRule = define $ \TypeCheck file -> do
     -- for files of interest on every keystroke
     typeCheckRuleDefinition hsc pm SkipGenerationOfInterfaceFiles
 
+data GetKnownFiles = GetKnownFiles
```

Comment:
Any reason not to host `GetKnownFiles` in `RuleTypes`? It could be useful in plugins